src/main/scala/com/cognite/sdk/scala/v1/transformations.scala

```diff
@@ -0,0 +1,175 @@
+    createdTime: Instant,
+    lastUpdatedTime: Instant,
+    owner: TransformationOwner,
+    ownerIsCurrentUser: Boolean,
+    hasSourceApiKey: Boolean,
+    hasDestinationApiKey: Boolean,
+    hasSourceOidcCredentials: Boolean,
+    hasDestinationOidcCredentials: Boolean,
+    lastFinishedJob: Option[JobDetails],
+    runningJob: Option[JobDetails],
+    externalId: String,
+    ignoreNullFields: Boolean,
+    dataSetId: Option[Long]
+) extends WithId[Long]
+    with WithCreatedTime
+    with ToCreate[TransformationCreate] {
+  override def toCreate: TransformationCreate =
+    TransformationCreate(
+      name = name,
+      query = Some(query),
+      destination = Some(destination),
+      conflictMode = Some(conflictMode),
+      isPublic = Some(isPublic),
+      sourceApiKey = None,
+      destinationApiKey = None,
+      sourceOidcCredentials = None,
+      destinationOidcCredentials = None,
+      externalId = externalId,
+      ignoreNullFields = ignoreNullFields,
+      dataSetId = dataSetId
+    )
+}
+
+final case class TransformBlacklistInfo(
+    reason: String,
+    createdTime: Instant
+)
+object TransformBlacklistInfo {
+  implicit val encoder: Encoder[TransformBlacklistInfo] = deriveEncoder[TransformBlacklistInfo]
+  implicit val decoder: Decoder[TransformBlacklistInfo] = deriveDecoder[TransformBlacklistInfo]
+}
+
+final case class TransformationOwner(user: String)
+object TransformationOwner {
+  implicit val ownerEncoder: Encoder[TransformationOwner] = deriveEncoder[TransformationOwner]
+  implicit val ownerDecoder: Decoder[TransformationOwner] = deriveDecoder[TransformationOwner]
+}
+
+sealed trait DestinationDataSource
+sealed case class GeneralDataSource(`type`: String) extends DestinationDataSource
+sealed case class RawDataSource(`type`: String, database: String, table: String)
+    extends DestinationDataSource
+
+sealed case class SequenceRowDataSource(`type`: String, externalId: String)
+    extends DestinationDataSource
+
+final case class JobDetails(
+    id: Int,
+    uuid: String,
+    transformationId: Int,
+    transformationExternalId: String,
+    sourceProject: String,
+    destinationProject: String,
+    destination: DestinationDataSource,
+    conflictMode: String,
+    query: String,
+    createdTime: Option[Long],
+    startedTime: Option[Long],
+    finishedTime: Option[Long],
+    lastSeenTime: Option[Long],
+    error: String,
+    ignoreNullFields: Boolean,
+    status: String
+)
+
+final case class FlatOidcCredentials(
+    clientId: String,
+    clientSecret: String,
+    scopes: Option[String],
+    tokenUri: String,
+    cdfProjectName: String,
+    audience: Option[String] = None
+)
+
+object FlatOidcCredentials {
+  implicit val encoder: Encoder[FlatOidcCredentials] = deriveEncoder[FlatOidcCredentials]
+  implicit val decoder: Decoder[FlatOidcCredentials] = deriveDecoder[FlatOidcCredentials]
+
+  implicit class ToClientCredentialConverter(flatCredential: FlatOidcCredentials) {
```

Comment:
We have avoided implicits classes in this SDK and I think we should continue to do so. It's better to have `FlatOidcCredentials.apply(clientCredentials: ClientCredentials)`, and a
```scala
final case class FlatOidcCredentials(...) {
  def clientCredentials(): ClientCredentials = ...
}
```
I'm not sure we should expose this to the SDK user at all, in fact. Would it be possible to use `ClientCredentials` in the SDK's API, and do the necessary conversions under the hood?

Reply:
I removed `FlatOidcCredentials` and use `ClientCredentials` instead directly, added a custom encoder for it, we don't need the decoder for now